src/app/integrations/create-page/create-page.component.ts

```diff
@@ -20,27 +20,96 @@ export class IntegrationsCreatePage implements OnInit, OnDestroy {
   integration: Observable<Integration>;
   private readonly loading: Observable<boolean>;
 
-  subscription: Subscription;
+  integrationSubscription: Subscription;
+  routeSubscription: Subscription;
+  childRouteSubscription: Subscription;
+  flowSubscription: Subscription;
+  urls: UrlSegment[];
+  canContinue = false;
+  position: number;
 
-  constructor(private store: IntegrationStore,
+  constructor(
+    private currentFlow: CurrentFlow,
+    private store: IntegrationStore,
     private route: ActivatedRoute,
-    private router: Router) {
+    private router: Router,
+    ) {
     this.integration = this.store.resource;
     this.loading = this.store.loading;
   }
 
+  cancel() {
+    this.router.navigate(['/integrations']);
+  }
+
+  finish() {
+    this.router.navigate(['/integrations']);
+  }
+
+  continue() {
+    const child = this.getCurrentChild();
+    switch (child) {
+      case 'connection-select':
+        this.router.navigate(['connection-configure', this.position], { relativeTo: this.route });
```

Comment:
I wasn't certain if that would be the case, so I wanted to cater for it, I can certainly envision going directly to the connection edit page in an integration for example, etc.